examples/demo-stf/src/sov-cli/native.rs

```diff
@@ -0,0 +1,487 @@
+        println!(
+            "private key written to path: {}",
+            path.into_os_string().into_string().unwrap()
+        );
+        Ok(())
+    }
+}
+
+impl SerializedTx {
+    fn new<P: AsRef<Path>>(
+        sender_priv_key_path: P,
+        module_name: &str,
+        call_data_path: P,
+        nonce: u64,
+    ) -> anyhow::Result<SerializedTx> {
+        let sender_priv_key = Self::deserialize_priv_key(sender_priv_key_path)?;
+        let sender_address = sender_priv_key.pub_key().to_address();
+        let message = Self::serialize_call_message(module_name, call_data_path)?;
+        println!("{:?}", message);
+        let tx = Transaction::<C>::new_signed_tx(&sender_priv_key, message, nonce);
+
+        Ok(SerializedTx {
+            raw: RawTx {
+                data: tx.try_to_vec()?,
+            },
+            sender: sender_address,
+        })
+    }
+
+    fn new_from_call<P: AsRef<Path>>(
+        sender_priv_key_path: P,
+        call_data: Vec<u8>,
+        nonce: u64,
+    ) -> anyhow::Result<SerializedTx> {
+        println!("{:?}", call_data);
+        let sender_priv_key = Self::deserialize_priv_key(sender_priv_key_path)?;
+        let sender_address = sender_priv_key.pub_key().to_address();
+        let tx = Transaction::<C>::new_signed_tx(&sender_priv_key, call_data, nonce);
+
+        Ok(SerializedTx {
+            raw: RawTx {
+                data: tx.try_to_vec()?,
+            },
+            sender: sender_address,
+        })
+    }
+
+    fn deserialize_priv_key<P: AsRef<Path>>(
+        sender_priv_key_path: P,
+    ) -> anyhow::Result<DefaultPrivateKey> {
+        let priv_key_data = std::fs::read_to_string(&sender_priv_key_path).with_context(|| {
+            format!(
+                "Failed to read private key from {:?}",
+                sender_priv_key_path.as_ref()
+            )
+        })?;
+
+        let sender_priv_key_data = serde_json::from_str::<PrivKeyAndAddress>(&priv_key_data)?;
+
+        Ok(DefaultPrivateKey::from_hex(
+            &sender_priv_key_data.hex_priv_key,
+        )?)
+    }
+
+    fn serialize_call_message<P: AsRef<Path>>(
+        module_name: &str,
+        call_data_path: P,
+    ) -> anyhow::Result<Vec<u8>> {
+        let call_data = std::fs::read_to_string(&call_data_path).with_context(|| {
+            format!(
+                "Failed to read call data from {:?}",
+                call_data_path.as_ref()
+            )
+        })?;
+        cmd_parser(module_name, &call_data)
+    }
+}
+#[cfg(feature = "native")]
+pub fn main() {
+    let cli = Cli::parse();
+    match cli.command {
+        Commands::SerializeCall {
+            sender_priv_key_path,
+            module_name,
+            call_data_path,
+            nonce,
+        } => {
+            let serialized =
+                SerializedTx::new(&sender_priv_key_path, &module_name, &call_data_path, nonce)
+                    .unwrap_or_else(|e| panic!("Call message serialization error: {}", e));
```

Comment:
nit: Since this is a user-facing tool, I don't think we want to panic at all if possible. Why don't we change the signature to `fn main() -> Result<(), anyhow::Error>`

Reply:
+1